js/profcond.js

```diff
@@ -273,6 +273,9 @@
       el = $('#' + condition.id);
     } else if (typeof condition.selector != 'undefined') {
       el = $(condition.selector);
+    } else if (typeof condition.label != 'undefined') {
+      condition.id = CRM.$("label:contains(" + condition.label + ")").attr('for');
```

Comment:
This seems like it could match more broadly than intended. E.g., if `condition.label = 'Foo'`, this would match elements with label 'FooBar', 'BarFoo', etc. Any chance you could tighten this to be an exact match?